src/Twig/Runtime/ImageRuntime.php

```diff
@@ -74,13 +75,22 @@ public function image($fileName = null, $width = null, $height = null, $crop = n
             $fileName = isset($fileName['filename']) ? $fileName['filename'] : (isset($fileName['file']) ? $fileName['file'] : '');
         }
 
+        // If _no_ filename is given, return nothing.
         if (!$fileName) {
             return null;
         }
 
-        $file = $this->filesystemMatcher->getFile($fileName);
-
-        $url = $file->url();
+        try {
+            $file = $this->filesystemMatcher->getFile($fileName);
+            $url = $file->url();
+        } catch (FileNotFoundException $e) {
+            // If a non-existing filename is given, we either re-throw the Exception (in DEV) or defer to the
+            // Thumbnailer, so the user sees the "404 image".
+            if ($this->config->get('general/debug')) {
```

Comment:
It would be better to use `$twigEnv->isDebug()` instead here.

Reply:
See #7122